src/pages/Compare/index.tsx

```diff
@@ -127,6 +132,19 @@ export class CompareBase extends React.Component<Props> {
     }
 
     if (version && currentVersionId !== version.id) {
+      if (currentVersionId) {
+        dispatch(versionsActions.restoreExpandedPath({ version }));
```

Comment:
It's apparent to me that `expandedPaths` should not be stored on the version object because 1. it might relate to multiple versions and 2. we need to persist it across version changes. However, that would be a big change to make.

As for making a smaller change towards the goal, do you think some reducer could just store `lastExpandedPaths` any time `expandedPaths` gets modified? Maybe that would allow us to restore the expanded paths either more easily or by not dispatching a specific `restoreExpandedPath` action at all? Just some thoughts.

Reply:
It makes more sense to put `expandedPaths` in `fileTree` object, rather than in `version`.
I can try to add `lastExpandedPaths` in `fileTree` to store the expanded paths, and let the action `buildTree` restore the expanded path. This might work.

Reply:
Yeah, sounds good. I would consider the `lastExpandedPaths` approach as temporary just to get the specific feature you need working but that's fine. In other words, changing all the existing code that relies on `version.expandedPaths` should remain as is, if possible, just for the first pass.

Reply:
In refactoring, I found it is much easier to move `expandedPaths` into `fileTree` than creating `lastExpandedPaths`. So I decided to try this approach first. I pushed the updates. (But it still needs a lot changes for tests...)

I am not sure about how to persist `selectedPath` and `visibleSelectedPath` when changing new version id. They should be moved out of `version` (maybe should be in `fileTree` the same as `expandedPath`?). But I feel that moving all three keys will make this patch too big. Could we leave `selectedPath/visibleSelectedPath` for another patch and let this patch only about `expandedPaths`?

Reply:
> Could we leave `selectedPath/visibleSelectedPath` for another patch and let this patch only about `expandedPaths`?

Yeah, definitely. Anything you can do to keep patches small and contained is a good thing.

I'll take a closer look at this shortly, probably early next week.